hoot-core/src/main/cpp/hoot/core/algorithms/alpha-shape/AlphaShape.cpp

```diff
@@ -328,38 +301,40 @@ std::shared_ptr<OsmMap> AlphaShape::_toOsmMap()
   return result;
 }
 
-std::shared_ptr<Geometry> AlphaShape::toGeometry()
+double AlphaShape::_collectValidFaces(const double alpha, std::vector<GeometryPtr>& faces, Envelope& e)
 {
-  LOG_DEBUG("Traversing " << _pDelauneyTriangles->getFaces().size() << " faces...");
-
-  // create a vector of all faces
-  vector<std::shared_ptr<Geometry>> tmp, tmp2;
-  Envelope e;
   double preUnionArea = 0.0;
-  int i = 0;
-  int numPolys = 0;
   for (FaceIterator fi = _pDelauneyTriangles->getFaceIterator();
        fi != _pDelauneyTriangles->getFaceEnd(); fi++)
   {
     const Face& f = *fi;
-    i++;
-    if (_isInside(f))
+    if (_isInside(alpha, f))
     {
-      std::shared_ptr<Polygon> p = _convertFaceToPolygon(f);
+      GeometryPtr p = _convertFaceToPolygon(f);
       LOG_VART(p->getArea());
-      tmp.push_back(p);
+      faces.push_back(p);
       e.expandToInclude(p->getEnvelopeInternal());
       preUnionArea += p->getArea();
-      numPolys++;
     }
   }
-  LOG_VARD(numPolys);
+  return preUnionArea;
+}
```

Comment:
Consolidated the pre-union checks so it is easier to call multiple times while calculating an alpha value.